prov/efa/src/efa_hmem.c

```diff
@@ -409,42 +372,18 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
  */
 int efa_domain_hmem_info_init_all(struct efa_domain *efa_domain)
 {
-	int ret, err;
+	int ret = 0;
+	enum fi_hmem_iface ifaces[4] = {FI_HMEM_SYSTEM, FI_HMEM_CUDA,
+	                                FI_HMEM_NEURON, FI_HMEM_SYNAPSEAI};
```

Comment:
Use the predefined static array from `efa_hmem.h`

https://github.com/ofiwg/libfabric/blob/5b08723bf29a914d045554c0942e6d058a58397e/prov/efa/src/efa_hmem.h#L16-L22